third_party/silabs/silabs_arm.gni

```diff
@@ -14,7 +14,7 @@
 
 import("silabs_board.gni")
 
-if (silabs_family == "efr32mg12" || silabs_family == "SiWx917") {
+if (silabs_family == "efr32mg12" || wifi_soc == true) {
```

Comment:
I would actually leave it as is. Maybe we will have another WIFI_SOC that won't be based on a cortex-M4 maybe a M33, or something else, who knows? that's not really future proof and constitute a regression.